src/typechecker.erl

```diff
@@ -3135,15 +3135,15 @@ add_types_pats([Pat | Pats], [Ty | Tys], TEnv, VEnv, PatTysAcc, UBoundsAcc, CsAc
            constraints:constraints()}.
 add_type_pat({var, _, '_'}, Ty, _TEnv, VEnv) ->
     {Ty, Ty, VEnv, constraints:empty()};
-add_type_pat({var, P, A}, Ty, TEnv, VEnv) ->
+add_type_pat({var, _, A} = Var, Ty, TEnv, VEnv) ->
     %% TODO: In a fun clause, A is always free, but not in e.g. case clauses
     case VEnv of
         #{A := VarTy} ->
             case subtype(VarTy, Ty, TEnv) of
                 {true, Cs} ->
                     {type(none), VarTy, VEnv, Cs};
                 false ->
-                    throw({type_error, tyVar, P, A, VarTy, Ty})
+                    throw({type_error, Var, VarTy, Ty})
```

Comment:
Just to be clear, I don't like the type error for the pattern to be the same as for the expression. (I know this may be out of the scope of this change, but then I think we should add a new task for *Unify type errors for patterns*.)